Sources/TestingMacros/Support/EffectfulExpressionHandling.swift

```diff
@@ -69,6 +69,19 @@ func findEffectKeywords(in node: some SyntaxProtocol, context: some MacroExpansi
   return effectFinder.effectKeywords
 }
 
+extension Sequence<Syntax> {
+  /// The syntax nodes in this sequence which are effectful expressions, such as
+  /// those for `try` or `await`.
+  var effectExpressions: some Sequence<Syntax> {
+    filter { node in
```

Comment:
I'm not certain what the compiler and swift-syntax are doing exactly, but this could still be incorrect:

```swift
// bad comment
try {
  #expect(...)
}()
```

This specific example is contrived, but do you see what I mean? I think you basically want to inspect the suffix of the lexical context sequence up to the first non-expr (and non-closure!) node.

Reply:
The part about only taking the _suffix_ of matching nodes absolutely makes sense. Implemented that in latest commit.

To err on the conservative/safe side right now I'm going to continue filtering to nodes of explicitly-specified types, rather than generalizing it to any `ExprSyntax`, and subtracting known-bad exprs like closures. I figure we can widen this over time, and it's primarily `try`/`await` that I care about at the moment.

Reply:
Makes sense!

Reply:
Also consider `UnsafeExprSyntax`!